services/alertmanager/alertmanager.go

```diff
@@ -130,6 +131,40 @@ receivers:
 		err = ioutil.WriteFile(alertmanagerBaseConfigPath, []byte(defaultBase), 0644) //nolint:gosec
 		svc.l.Infof("%s created: %v.", alertmanagerBaseConfigPath, err)
 	}
+
+	var content []byte
+	_, err = os.Stat(alertmanagerBaseConfigPath)
+	svc.l.Debugf("%s status: %v", alertmanagerBaseConfigPath, err)
+	if err == nil {
+		svc.l.Infof("checking %s content", alertmanagerConfigPath)
+		content, err = ioutil.ReadFile(alertmanagerConfigPath)
+		if err != nil {
+			svc.l.Errorf("Failed to load alertmanager config %s: %s", alertmanagerConfigPath, err)
+		}
+	}
+
+	// copy the base config if `/etc/alertmanager.yml` is not present or
+	// is already present but does not have any config.
+	if string(content) == "---\n" || os.IsNotExist(err) {
+		var cfg alertmanager.Config
+		buf, err := ioutil.ReadFile(alertmanagerBaseConfigPath)
+		if err != nil {
+			svc.l.Errorf("Failed to load alertmanager base config %s: %s", alertmanagerBaseConfigPath, err)
```

Comment:
Shouldn't we call `return` here?

Reply:
kept logging the same and added naked `return`s in case of errors to keep the error handling style consistent with other methods in this file

let me know if you wanted something different like returning the error from `generateBasicConfig` and logging/handling it in the `Run` method instead